synapse/handlers/room_member.py

```diff
@@ -683,3 +706,24 @@ def forget(self, user, room_id):
 
         if membership:
             yield self.store.forget(user_id, room_id)
+
+    @defer.inlineCallbacks
+    def _is_host_in_room(self, current_state_ids):
+        # Have we just created the room, and is this about to be the very
+        # first member event?
+        create_event_id = current_state_ids.get(("m.room.create", ""))
+        if len(current_state_ids) == 1 and create_event_id:
+            defer.returnValue(self.hs.is_mine_id(create_event_id))
+
+        for (etype, state_key), event_id in current_state_ids.items():
```

Comment:
This could get interesting if there were a lot of users on this server that had left the room. But I don't think we need to worry too much about it.